File: src/main/java/com/example/demo/model/WebAuthnCredential.java
package com.example.demo.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

/**
 * WebAuthn registered credential.
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WebAuthnCredential {
    private Long id;
    private String username;
    private byte[] credentialId;
    private byte[] publicKeyCose;
    private Long signCount;
    private String transports;
    private String attestationType;
    private String aaguid;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: src/main/kotlin/com/example/demo/model/WebAuthnCredential.kt
package com.example.demo.model

import java.time.OffsetDateTime

/** WebAuthn登録済みクレデンシャル */
data class WebAuthnCredential(
    val id: Long? = null,
    val username: String,
    val credentialId: ByteArray,
    val publicKeyCose: ByteArray,
    val signCount: Long,
    val transports: String?,
    val attestationType: String?,
    val aaguid: String?,
    val createdAt: OffsetDateTime? = null,
    val updatedAt: OffsetDateTime? = null,
)

